src/Symfony/Bundle/SecurityBundle/DependencyInjection/MainConfiguration.php

```diff
@@ -418,6 +418,15 @@ private function addProvidersSection(ArrayNodeDefinition $rootNode): void
     private function addPasswordHashersSection(ArrayNodeDefinition $rootNode): void
     {
         $rootNode
+            ->validate()
+                ->always(function (array $v): array {
+                    if (empty($v['password_hashers'])) {
+                        trigger_deprecation('symfony/framework-bundle', '6.4', 'Not setting the "security.password_hashers" config option is deprecated. It will default to "Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface: auto" in 7.0.');
```

Comment:
I'm not sure about this deprecation. Default values for a prototyped array node are not merged with provided config. So as soon as any config file (or config prepended by a bundle) adds some other config for password hashers, the config for PasswordAuthenticatedUserInterface would be lost.

I'd suggest keeping that one in the recipe.

Reply:
They would be merged if we didn't call `performNoDeepMerging`, isn't it?
Why don't we perform deep merging BTW?

Reply:
Anyway, let's drop this, it's not needed to deprecate not setting this

Reply:
:shrug: a5cfc2207c09f6182333b3b812673202701c33f3 "some bug fixes in DIC config classes"

Reply:
@nicolas-grekas no. deep merging is about the various configs received as input, not the defaults